app/src/main/java/nl/tudelft/b_b_w/model/block/GenesisBlock.java

```diff
@@ -27,11 +29,32 @@ public GenesisBlock(BlockData blockData) throws HashException {
         super(blockData);
 
         // verify integration and integrity
-        if (BuildConfig.DEBUG && blockData.getBlockType() != BlockType.GENESIS
-                || blockData.getSequenceNumber() != firstSequenceNumber
-                || !blockData.getPreviousHashChain().equals(notAvailable)
-                || !blockData.getPreviousHashSender().equals(notAvailable)) {
-            throw new AssertionError("Invalid genesis block " + this.toString());
+        if (BuildConfig.DEBUG) {
+            if (blockData.getBlockType() != BlockType.GENESIS
+                    || blockData.getSequenceNumber() != FIRST_SEQUENCE_NUMBER
+                    || !blockData.getPreviousHashChain().equals(notAvailable)
+                    || !blockData.getPreviousHashSender().equals(notAvailable)) {
+                throw new AssertionError("Invalid genesis block " + this.toString());
+            }
         }
     }
+
+    /**
+     * Easy method for genesis block creation.
+     * TODO: move this to constructor of genesis block
```

Comment:
This will happen in another PR because it is a large refactor. The BlockData class will be rewritten to a BlockBuilder class with properties moved to the Block itself: this method now exists because you cannot create an object and then immediately passing it to the super constructor, which is necessary with BlockData.